File: python/7 kyu/252.py
# https://www.codewars.com/kata/57238e345b21bb8c4b0000fc/train/python

# Passed

def i_speak_french(sentence):
    sentences = [s for s in sentence.split(".") if s]
    modified_sentences = []
    for sentence_to_process in sentences:
        words = [w for w in sentence_to_process.split(" ") if w]
        modified_words = ["baguette"] * len(words) + ["Encore!"]
        modified_words[0] = modified_words[0].title()
        modified_sentence = " ".join(modified_words)
        modified_sentences.append(modified_sentence)
        
    processed_sentence = " ".join(modified_sentences)
    return processed_sentence

output = i_speak_french("I speak French.")
print(output)

File: nodejs/7 kyu/252.js
// https://www.codewars.com/kata/57d2807295497e652b000139/train/javascript

// Passed

function averages(numbers) {
  let averagesArr = [];
  if (numbers === null || [1, 0].includes(numbers.length)) {
    return averagesArr;
  }

  for (let idx = 0; idx < numbers.length - 1; idx++) {
    let average = (numbers[idx] + numbers[idx + 1]) / 2;
    averagesArr.push(average);
  }

  return averagesArr;
}

const output = averages([1, 3, 5, 1, -10]);
console.log(output);